External/SPMmodified/spm_load.m

```diff
@@ -162,11 +164,43 @@
 end
 d = textscan(S,'%s','Delimiter',delim);
 
-if rem(numel(d{1}),N)
+% EXPLOREASL HACK: detect potentially erroneous empty cells
+illegalEmptyCells = DetectIllegalEmptyCells(d, N);
+
+if ~isempty(illegalEmptyCells)
+    warning(['Found illegal empty cells in ' f ', trying to repair']);
+    d = FixIllegalEmptyCells(d, N);
```

Comment:
I don't like that you first detect, then run fix that does detection as the first thing.